doc/BUILD.rst

```diff
@@ -0,0 +1,730 @@
+for any of its supported operating systems, including Linux, macOS, Windows,
+Emscripten, Switch, and others.
+
+.. contents::
+
+Dependencies
+------------
+
+Build-Time Dependenices
+"""""""""""""""""""""""
+
+-  ``gcc`` or ``clang``
+-  meson >= 0.53.0 (0.56.2 recommended)
+-  Python >= 3.6
+-  `python-zstandard <https://github.com/indygreg/python-zstandard>`__ >= 0.11.1
+-  `python-docutils <https://pypi.org/project/docutils/>`__ (optional, for generating documentation)
+
+Run-Time Dependencies
+"""""""""""""""""""""
+
+Required
+''''''''
+
+-  OpenGL >= 3.3, or OpenGL ES >= 3.0
+-  SDL2 >= 2.0.10
+-  cglm >= 0.7.8
+-  libpng >= 1.5.0
+-  libwebpdecoder >= 0.5 or libwebp >= 0.5
+-  libzip >= 1.5.0 (>= 1.7.0 recommended)
+-  libzstd >= 1.4.0
+-  freetype2
+-  opusfile
+-  zlib
+
+Optional
+''''''''
+
+-  OpenSSL (for a better SHA-256 implementation; used in shader cache)
+-  SPIRV-Cross >= 2019-03-22 (for OpenGL ES backends)
+-  libshaderc (for OpenGL ES backends)
+-  `ANGLE <https://github.com/google/angle>`__ (useful for platforms with
+   flaky/non-existent OpenGL support, such as Windows)
+-  GameMode headers (Linux only; for automatic
+   `GameMode <https://github.com/FeralInteractive/gamemode>`__ integration)
+
+
+Built-In vs. System Dependencies
+""""""""""""""""""""""""""""""""
+
+Due to the wide array of platforms Taisei supports, the project packages many of
+its core dependencies inside the project using the
+`Meson dependency wrap system <https://mesonbuild.com/Wrap-dependency-system-manual.html>`__.
+This is used to pin specific versions of libraries such as `SDL2` to known-good
+versions, as well as assist with packaging and distribution on older and/or more
+esoteric systems (i.e: Emscripten).
+
+For convenience, ``meson`` will detect which packages are missing from your
+system and use its wrap dependency system to pull in what it can. The trade-off
+is that this can result in longer build times.
+
+For consistency, we tend to release Taisei using exclusively built-in packages.
+However, you can also use system dependencies as well. There's a tradeoff in
+consistency and reproduceability for speed and ease of use.
+
+This is controlled through the ``--wrap-mode`` flag with ``meson``. (More
+on that later.)
+
+Linux
+'''''
+
+On an Ubuntu or Debian-based distro, he following will install the mandatory
+tools for building Taisei.
+
+.. code:: sh
+
+   apt update
+   apt install meson cmake build-essential
+
+Beyond that, consult the *Optional Dependencies* list above. Many distros
+package compile-time system dependencies with ``*-dev`` (i.e: ``libsdl2-dev``).
+Search with your distro's package manager to install the correct libraries.
+
+If you're using built-in dependencies *only* for whatever reason, one
+non-obvious dependency is for Wayland-based systems, where you may need to
+install the Wayland development library to ensure SDL2 compiles for it
+correctly.
+
+.. code:: sh
+
+   apt install libwayland-dev
```

Comment:
It really goes way beyond that, e.g. if you're on X11 you need to install a bunch of X11 libs as well; if you want audio you need to install pulse or pipewire or whatever libs, etc… There's a good reason why I don't recommend building SDL as a subproject; if we really want to document it beyond "DO NOT DO THIS", it probably warrants a whole separate section.